It seems TRUE/FALSE/boolean definitions from various files
(libjpeg and dbus?) included from image.cpp conflict.

Define them first to resolve the conflicts.

diff -dur slim-1.3.6.orig/image.cpp slim-1.3.6/image.cpp
--- slim-1.3.6.orig/image.cpp	2013-10-02 00:38:05.000000000 +0200
+++ slim-1.3.6/image.cpp	2013-12-12 11:58:37.000000000 +0100
@@ -21,6 +21,11 @@
 
 using namespace std;
 
+#define TRUE 1
+#define FALSE 0
+#define boolean int
+#define HAVE_BOOLEAN 1
+
 #include "image.h"
 
 extern "C" {
